### ISSUE TITLE
Required Python Version Inconsistency

### ISSUE THREAD

**Issue by mlh758** [aws/aws-cli#3325]
The installation documentation for installing AWS CLI [here](https://docs.aws.amazon.com/cli/latest/userguide/installing.html) state that the required Python version is 2.6.5+

However, it installs [PyYaml](http://pyyaml.org/wiki/PyYAML) as a dependency and that requires Python 2.7.

It seems like it might be necessary to specify the version of PyYAML being pulled in to something else.

Edit: Although 3.12 was released 2 years ago. Surely someone else would have run into this problem by now?

**Comment by joguSD** [comment]
Is there an error or warning that you're getting when trying to run the cli on 2.6?

**Comment by lorengordon** [comment]
That page right at the top indicates it is "historical" which to me would mean it is not necessarily accurate. The project's setup.py still lists support for python 2.6:

* https://github.com/yaml/pyyaml/blob/master/setup.py#L30

And there are issues/pull requests discussing keeping or deprecating it, thus far landing in favor of keeping it. Though currently there is an active pull request where the discussion looks to be leaning towards deprecating py2.6:

* https://github.com/yaml/pyyaml/pull/105

**Comment by mlh758** [comment]
I was installing it with [Chef](https://supermarket.chef.io/cookbooks/cloudcli) which seems to just pip install the aws cli tool. While it was installing PyYAML a syntax error was thrown from a setup.py file and the message was saying I was using an unsupported version of Python.

That brought me to the PyYAML docs.

We're just upgrading Python to 3.4.5 since there wasn't anything keeping us on the older version, it's just what came on the VM's image so this isn't a major blocker for me.

**Comment by mlh758** [comment]
![image](https://user-images.githubusercontent.com/4331955/40145393-33a51674-5927-11e8-8adf-d1f6eb44e87b.png)

**Comment by joguSD** [comment]
The actual issue is not that `pyyaml` doesn't support 2.6, but the version of the `wheel` package being used to build the wheel for `pyyaml` doesn't support 2.6.

**Comment by mlh758** [comment]
Ah, I assumed it was part of pyyaml. Same sort of problem though, isn't it?

**Comment by lorengordon** [comment]
No, this is about the managing the local environment in which pip is executing. Pyyaml can't actually control that when you run `pip install`. It ends up being a user problem. You can `pip install 'wheel<0.30.0'` and then pyyaml ought to build just fine.

There are actually now three "setup" packages whose versions are important to control on py2.6, `setuptools<37`, `wheel<0.30.0`, and `pip<10`. All have deprecated support for py2.6 in their latest versions, so if you are running on py2.6, it's up to you to use versions that support the runtime.

**Comment by mlh758** [comment]
But if the package is trying to support older runtimes shouldn't it fix dependency versions that support that runtime as well?

In any case we upgraded Python to 3.4.3 and are getting the same error. So there has to be something else going on. It's working with chef in our lower environments and in our test VMs and I just verified that the Python versions (before bumping to 3.4.3), wheel versions, and `pyyaml` are the same as in our lower environments which built successfully.

Unless `pip` linked to the older install of Python in a way I didn't anticipate (I just aliased python3 over python after upgrading) there has to be something else going on.
